### ISSUE TITLE
WSL2 not starting despite Hyper-V and virtualization being enabled

### ISSUE THREAD

**Issue by WalshyDev** [microsoft/WSL#4930]
Background: I used to run WSL1 & WSL2 perfectly fine, I had to disable Hyper-V along with WSL a little bit ago. Ever since I disabled them in the past I haven't been able to enable everything properly again. Hyper-V features on enabled and services running, Virtualization is enabled in BIOS but yet... I cannot start WSL.

**Please fill out the below information:**
* Microsoft Windows [Version 10.0.19025.1051]

* What you're doing and what's happening:
```
C:\WINDOWS\system32>wsl
Please enable the Virtual Machine Platform Windows feature and ensure virtualization is enabled in the BIOS.
For information please visit https://aka.ms/wsl2-install
```

* What's wrong / what should be happening instead:
It should just go into my WSL2 VM, it is instead just spitting out: `Please enable the Virtual Machine Platform Windows feature and ensure virtualization is enabled in the BIOS.`

Features:
![Image of Windows features](https://blobs.are-pretty.sexy/9dAiVLg.png)

Services:
![Image of services](https://blobs.are-pretty.sexy/88iyzya.png)

Task Manager showing Virtualization enabled:
![Task Manager showing Virtualization enabled](https://blobs.are-pretty.sexy/TnAQdJQ.png)

* Strace of the failing command, if applicable: N/A

*I have attached the logman details (Started before running `wsl` and stopped after it failed):
[wsl-report.zip](https://github.com/microsoft/WSL/files/4252790/wsl-report.zip)

**Comment by DropNib** [comment]
Just to be sure, you do have the feature "Virtual Machine Platform" enabled, as in this picture?

<img width="450" alt="image" src="https://user-images.githubusercontent.com/32019589/75359211-5f08ef00-58b4-11ea-9936-a2b52f335679.png">

**Comment by ad-on-is** [comment]
Did you try `bcdedit /set hypervisorlaunchtype auto`

I disabled hyper-v with `bcdedit /set hypervisorlaunchtype off` to do some work in VMWare, and ebabled it with `auto` successfully back again.

**Comment by WalshyDev** [comment]
@DropNib Yep, check the summary under `wsl -l -v` output and above the strace output.

@ad-on-is Yep, already tried with no success.
```
Windows Boot Manager
--------------------
identifier              {bootmgr}
device                  partition=\Device\HarddiskVolume5
path                    \EFI\Microsoft\Boot\bootmgfw.efi
description             Windows Boot Manager
locale                  en-GB
inherit                 {globalsettings}
flightsigning           Yes
default                 {current}
resumeobject            {fa62a036-d7e2-11e9-8624-8aedbab3624a}
displayorder            {current}
toolsdisplayorder       {memdiag}
timeout                 0

Windows Boot Loader
-------------------
identifier              {current}
device                  partition=C:
path                    \WINDOWS\system32\winload.efi
description             Windows 10
locale                  en-GB
inherit                 {bootloadersettings}
recoverysequence        {3f7b4696-0c00-11ea-b600-ebdd96078e40}
displaymessageoverride  Recovery
recoveryenabled         Yes
isolatedcontext         Yes
flightsigning           Yes
allowedinmemorysettings 0x15000075
osdevice                partition=C:
systemroot              \WINDOWS
resumeobject            {fa62a036-d7e2-11e9-8624-8aedbab3624a}
nx                      OptIn
bootmenupolicy          Standard
hypervisorlaunchtype    Auto
```

**Comment by ad-on-is** [comment]
Stupid question, but, did you reboot Windows after enabling everything? 😅

**Comment by WalshyDev** [comment]
@ad-on-is Oh yes. There's been many many reboots during the process of trying to fix this over the last month. Even rebooted since last night when I posted this but yeah... nothing :/

**Comment by Gareth001** [comment]
I was having this same problem. I disabled AMD SVM, booted, then re-enabled AMD SVM and it worked again. Hope this helps

**Comment by onomatopellan** [comment]
First of all, if you are in Windows 10 Pro try to run an hyper-v VM or launch Windows Sandbox. If those don't work either then the problem is in hyper-v.

**Comment by moriarity9211** [comment]
> 
> 
> Did you try `bcdedit /set hypervisorlaunchtype auto`
> 
> I disabled hyper-v with `bcdedit /set hypervisorlaunchtype off` to do some work in VMWare, and ebabled it with `auto` successfully back again.

So, It is not possible to use VM and WSL2 at same time?

**Comment by benhillis** [comment]
@moriarity9211 - VMware Workstation 15.5 supports it.

https://blogs.vmware.com/workstation/2020/05/vmware-workstation-now-supports-hyper-v-mode.html

**Comment by ravesonly** [comment]
@benhillis 
How it's going to help on original issue, did anyone solved this original issue in Win 10 home edition successfully? Can you please elaborate valid steps?
Thanks in advance!

**Comment by soccerdude288** [comment]
> Did you try `bcdedit /set hypervisorlaunchtype auto`
> 
> I disabled hyper-v with `bcdedit /set hypervisorlaunchtype off` to do some work in VMWare, and ebabled it with `auto` successfully back again.

Mine was working fine till I started up an Android Studio Emulator last week. Since then I've been fighting to try and get wsl2 working again. BIOS was good, features were turned on, but no dice.

Finally tried the command above with a reboot and it worked!

**Comment by moriarity9211** [comment]
It works for me after update. I tried both bcdebit /set hypervisorlaunchtype auto & off it works but i have to restart my pc every time to update setting. After update it is working fine.

**Comment by benhillis** [comment]
VMWare Workstation 15.5.5 Pro supports using the [Microsoft Hypervisor Platform APIS](https://docs.microsoft.com/en-us/virtualization/api/).  You will need to be running that version or later.

WSL2 requires the "Virtual Machine Platform" optional component which is available on all SKUs.

**Comment by johannfenech** [comment]
This fixed it for me https://blog.johannfenech.com/fix-wslregisterdistribution-failed-with-error-0x80370102/

It took me days and two formats to figure out !

**Comment by kubichris** [comment]
> > Did you try `bcdedit /set hypervisorlaunchtype auto`
> > I disabled hyper-v with `bcdedit /set hypervisorlaunchtype off` to do some work in VMWare, and ebabled it with `auto` successfully back again.
> 
> Mine was working fine till I started up an Android Studio Emulator last week. Since then I've been fighting to try and get wsl2 working again. BIOS was good, features were turned on, but no dice.
> 
> Finally tried the command above with a reboot and it worked!

Yes, it work for wsl, but doesn't good for VirtualBox. 

WSL : bcdedit /set hypervisorlaunchtype auto + RESTART
VirtualBox: bcdedit /set hypervisorlaunchtype off + RESTART

**Comment by pariks90900** [comment]
> Did you try `bcdedit /set hypervisorlaunchtype auto`
> 
> I disabled hyper-v with `bcdedit /set hypervisorlaunchtype off` to do some work in VMWare, and ebabled it with `auto` successfully back again.

Mine worked after this command

wsl doesn't work with other software using virtualization 
I installed Andy Android Emulator and then this error code '0x80370102' appeared 
maybe some VMs changes the windows features and wsl don't work after that

**Comment by ZayenJS** [comment]
> Did you try `bcdedit /set hypervisorlaunchtype auto`
> 
> I disabled hyper-v with `bcdedit /set hypervisorlaunchtype off` to do some work in VMWare, and ebabled it with `auto` successfully back again.

I was having the same problem, this worked great, thanks a lot !

**Comment by mrivard** [comment]
@WalshyDev, at 2:58 in, [this YouTuber](https://youtu.be/_fntjriRe48)  says to *not* enable Windows Hypervisor Platform when adding WSL2. He doesn't say why, though.

**Comment by feland207** [comment]
> > Did you try `bcdedit /set hypervisorlaunchtype auto`
> > I disabled hyper-v with `bcdedit /set hypervisorlaunchtype off` to do some work in VMWare, and ebabled it with `auto` successfully back again.
> 
> So, It is not possible to use VM and WSL2 at same time?

I quote this question. Is any way to have both WSL2 and VM ?

**Comment by yiqiangjizhang** [comment]
> > > Did you try `bcdedit /set hypervisorlaunchtype auto`
> > > I disabled hyper-v with `bcdedit /set hypervisorlaunchtype off` to do some work in VMWare, and ebabled it with `auto` successfully back again.
> > 
> > 
> > So, It is not possible to use VM and WSL2 at same time?
> 
> I quote this question. Is any way to have both WSL2 and VM ?

I quote this reply as well, I want to have both WSL and VM working, is this possible? Thanks to all community for helping out!

**Comment by zhangxj19** [comment]
> Did you try `bcdedit /set hypervisorlaunchtype auto`
> 
> I disabled hyper-v with `bcdedit /set hypervisorlaunchtype off` to do some work in VMWare, and ebabled it with `auto` successfully back again.
It really works if you ever tried android emulator.

**Comment by yiqiangjizhang** [comment]
> > Did you try `bcdedit /set hypervisorlaunchtype auto`
> > I disabled hyper-v with `bcdedit /set hypervisorlaunchtype off` to do some work in VMWare, and ebabled it with `auto` successfully back again.
> > It really works if you ever tried android emulator.

Yeah it worked finally!

I used:
```
Activate VirtualBox (Disable Hyper-V):
bcdedit /set hypervisorlaunchtype off

Activate WSL (Enable Hyper-V):
bcdedit /set hypervisorlaunchtype auto
```

to enable and disable Hyper V so I can run WSL or VM. However, the PC needs restart

**Comment by Sorin-M-Oprea** [comment]
I have Hyper-V, Virtual Machine Platform and all the required features enabled, but this still doesn't work for me even after setting the bcd entry to auto.

**Comment by ElCosmoXD** [comment]
I have the same issue, I have all the required features enabled (with the virtualization enabled in the BIOS). I don't have any emulator like the Android simulator or anything like that. I tried everything but nothing solves this issue.
![image](https://user-images.githubusercontent.com/37759352/146696621-08bf385b-7fcb-410d-9773-149f09fcf333.png)

**Comment by husniadil** [comment]
Had no problem last week. But this is now happening to me. I didn't touch anything about the VirtualMachinePlatform. I didn't touch any BIOS settings. I'm relying on WSL2 for my work and now suddenly not working.

**Comment by fellipec** [comment]
> Had no problem last week. But this is now happening to me. I didn't touch anything about the VirtualMachinePlatform. I didn't touch any BIOS settings. I'm relying on WSL2 for my work and now suddenly not working
> 
> I tried:
> 
> * reinstall WSL via Turn on/off features
> * run dism thingy to disable and enable WSL2 and Virtual Machine Platform
> * disable + re-enable the VT-d and Intel Vietualization Technology
> * run bcedit thingy to auto, and off as well
> * enable / disable hypervisor
> * uninstall docker desktop
> * uninstall windows subsystem for android
> * reset network adapters
> * reset winsock
> * restart the computer many times
> * check windows event viewer: no error related to wsl
> * subscribe to beta channel
> 
> Nothing worked for me.

Same thing as you.

**Comment by husniadil** [comment]
I ended up re-installing Windows and tried to reinstall all programs one by one that were previously installed in my system. Took several hours for me (re-installed Windows multiple times) until I figured out which program did ruin the WSL.

Turned out that the culprit was [spacedesk](https://www.spacedesk net?ref=wsl-error). So, my solution was to **not** install spacedesk after re-installing, and WSL2 run smoothly without problems.

**Comment by fellipec** [comment]
I got so frustrated today that I did the same. I'm in the process of reinstall the things. But before I choose to format the PC, , I was suspecting spacedesk too, but I just disabled it, and did not restart too, and dismissed because it worked before.

I'll not install that again.

**Comment by ElCosmoXD** [comment]
> I ended up re-installing Windows and tried to reinstall all programs one by one that were previously installed in my system. Took several hours for me (re-installed Windows multiple times) until I figured out which program did ruin the WSL.
> 
> Turned out that the culprit was [spacedesk](https://www.spacedesk.net?ref=wsl-error). So, my solution was to **not** install spacedesk after re-installing Windows.

I also installed spacedesk before the error but at the same time I uninstalled it and I installed [Duet Display](https://duetdisplay.com) so I don't know which program was the culprit.

**Comment by digitalcora** [comment]
I'm also having this issue seemingly as a result of installing Spacedesk. Trying to start, export, or change the version of an existing WSL2 VM gives the "Please enable the Virtual Machine Platform Windows feature" error message. I've tried every fix suggested here and in [this thread](https://www.reddit.com/r/bashonubuntuonwindows/comments/p87s4y/wsl2_on_windows_10_not_booting_with_please_enable/) but with no success. Thankfully WSL1 still works, after I reinstalled the distro.

* Windows 10.0.19044.1415
* WSL kernel version 5.10.60.1
* Spacedesk version 1.0.25

**Comment by Martin-SF** [comment]
For me nothing described here or elsewhere worked.
Finally it fixed it for me by doing a update installation through the MediaCreationTool. Windows looks and behaves exact the same but wsl finally is working.

One Tip for anyone wondering about the output of the Intel® Processor Identification Utility. It still shows for me an X for Intel Virtualisation Technology and for EPT /SLAT support. But my wsl2 and Windows sandbox is working again.

**Comment by daniel-iroka** [comment]
> I got so frustrated today that I did the same. I'm in the process of reinstall the things. But before I choose to format the PC, , I was suspecting spacedesk too, but I just disabled it, and did not restart too, and dismissed because it worked before.
> 
> I'll not install that again.

Wow! Yes! I also had this issue after installing spacedesk! But even after deleting spacedesk it still didn't work. Please what else can I do apart from reinstalling windows?? Or will reinstalling wsl2 help?

**Comment by daniel-iroka** [comment]
> I got so frustrated today that I did the same. I'm in the process of reinstall the things. But before I choose to format the PC, , I was suspecting spacedesk too, but I just disabled it, and did not restart too, and dismissed because it worked before.
> 
> I'll not install that again.

wait, when you said you "disabled spacedesk", did you mean you uninstalled it and did not restart because I uninstalled it and restarted my pc but it is still not working.

**Comment by Martin-SF** [comment]
> > I got so frustrated today that I did the same. I'm in the process of reinstall the things. But before I choose to format the PC, , I was suspecting spacedesk too, but I just disabled it, and did not restart too, and dismissed because it worked before.
> > I'll not install that again.
> 
> wait, when you said you "disabled spacedesk", did you mean you uninstalled it and did not restart because I uninstalled it and restarted my pc but it is still not working.

I read that about spacedesk just now. i updated it recently and then wsl2 stopped working. I updated like 10 programs in the same time so I didnt really suspeted that spacedesk maybe have something to do with wsl failing. But I guess thats an important clue, spacedesk is intefering with wsl / virtualisation on windows. 

@daniel-iroka I would suggest you the update installation with the MediaCreationTool. It reinstalled windows and after 30 min my ubuntu distro ran again.... iam so happy, but also sad about 2 days of lost time, because of such an ackward bug

**Comment by fellipec** [comment]
> > I got so frustrated today that I did the same. I'm in the process of reinstall the things. But before I choose to format the PC, , I was suspecting spacedesk too, but I just disabled it, and did not restart too, and dismissed because it worked before.
> > I'll not install that again.
> 
> wait, when you said you "disabled spacedesk", did you mean you uninstalled it and did not restart because I uninstalled it and restarted my pc but it is still not working.

No, I just disabled the driver in the main window. I have no screenshots and I'll not install it again to do one, sorry.
If you fully uninstalled, reboot and `sfc /scannow` it don't fix the issue, MAYBE it have a rootkit-like behavior of some sorts. 
As far as I know what we have in common here is having installed Spacedesk before our WSL2/Hypervisor gone corrupted.
Also, I couldn't use VMWare too before formatting my machine.

**Comment by daniel-iroka** [comment]
> > > I got so frustrated today that I did the same. I'm in the process of reinstall the things. But before I choose to format the PC, , I was suspecting spacedesk too, but I just disabled it, and did not restart too, and dismissed because it worked before.
> > > I'll not install that again.
> > 
> > 
> > wait, when you said you "disabled spacedesk", did you mean you uninstalled it and did not restart because I uninstalled it and restarted my pc but it is still not working.
> 
> I read that about spacedesk just now. i updated it recently and then wsl2 stopped working. I updated like 10 programs in the same time so I didnt really suspeted that spacedesk maybe have something to do with wsl failing. But I guess thats an important clue, spacedesk is intefering with wsl / virtualisation on windows.
> 
> @daniel-iroka I would suggest you the update installation with the MediaCreationTool. It reinstalled windows and after 30 min my ubuntu distro ran again.... iam so happy, but also sad about 2 days of lost time, because of such an ackward bug

I have few more questions please. What about the normal settings update as well? Will that also work and will I also loose data if i update through the media tool?? Also, how will I know what media tool link to follow can you help me with a link please. My OS buid is 19043

**Comment by husniadil** [comment]
Re-installing Windows with "Keep my personal files" settings worked for me.

I you uninstall spacedesk using spacedesk installer before re-installing Windows, the driver will not be brought back to your new Windows installation.

But if you don't uninstall spacedesk, Windows will still bring spacedesk over your new Windows installation, BUT somehow they work together (WSL and spacedesk), then again don't use the spacedesk installer to install it again.

**Comment by daniel-iroka** [comment]
> Re-installing Windows with "Keep my personal files" settings worked for me.
> 
> I you uninstall spacedesk using spacedesk installer before re-installing Windows, the driver will not be brought back to your new Windows installation.
> 
> But if you don't uninstall spacedesk, Windows will still bring spacedesk over your new Windows installation, BUT somehow they work together (WSL and spacedesk), then again don't use the spacedesk installer to install it again.

Okay. Thank you

**Comment by yangmeishu** [comment]
> Finally it fixed it for me by doing an update installation "keep my personal files and programs" through the MediaCreationTool. Wsl (and Windows sandbox) is finally working again.
> 
> One tip for anyone wondering about the output of the Intel® Processor Identification Utility. It still shows for me an X (for not enabled) for Intel Virtualisation Technology and EPT/SLAT support. But my wsl2 and Windows sandbox is working fine. (i5-10600k)

This problem bothers me too much time and energy. Before I saw this post, I had no reason to connect the two software wsl and spacedesk. Similarly, I also installed spacedesk on my computer, and it was updated recently. Finally see your operation. The problem is solved

**Comment by daniel-iroka** [comment]
> > Finally it fixed it for me by doing an update installation "keep my personal files and programs" through the MediaCreationTool. Wsl (and Windows sandbox) is finally working again.
> > One tip for anyone wondering about the output of the Intel® Processor Identification Utility. It still shows for me an X (for not enabled) for Intel Virtualisation Technology and EPT/SLAT support. But my wsl2 and Windows sandbox is working fine. (i5-10600k)
> 
> This problem bothers me too much time and energy. Before I saw this post, I had no reason to connect the two software wsl and spacedesk. Similarly, I also installed spacedesk on my computer, and it was updated recently. Finally see your operation. The problem is solved

wait, so you mean that spacedesk is now working well with wsl??

**Comment by yangmeishu** [comment]
No, if spacedesk is installed, wsl cannot be started. It can be repaired according to your method. After reinstalling the system, wsl and spacedesk coexist

**Comment by yangmeishu** [comment]
> 

No, if spacedesk is installed, wsl cannot be started. It can be repaired according to your method. After reinstalling the system, wsl and spacedesk coexist

**Comment by daniel-iroka** [comment]
> > 
> 
> No, if spacedesk is installed, wsl cannot be started. It can be repaired according to your method. After reinstalling the system, wsl and spacedesk coexist

Oh that's nice. I uninstalled spacedesk though because I was told wsl won't still work if I reinstall it with windows

**Comment by yangmeishu** [comment]
> l

Without uninstalling spacedesk, after reinstalling windows, it can be used normally. But I think it would be better to use a lower version of spacedesk

**Comment by grzegorztomasiak** [comment]
+1 
I am also looking for a solution. After installing spacedesk, wsl stopped working and re-enabling it does not do anything.

**Comment by Nachtalb** [comment]
There is a thread open over at the spacedesk forum referencing this thread here so for cross reference sake: https://www.spacedesk.net/de/forums/topic/wsl2-not-starting-after-spacedesk-installation/

If any of you have more information about this that could help the spacedesk guys, please tell them :)

**Comment by Darkan35** [comment]
I found this.
Wsl is not started because Hyper V service is not started and I can't started manually this.
I Try to install an old version but Wsl don't start.

**Comment by Lyndonpett** [comment]
> Did you try `bcdedit /set hypervisorlaunchtype auto`
> 
> I disabled hyper-v with `bcdedit /set hypervisorlaunchtype off` to do some work in VMWare, and ebabled it with `auto` successfully back again.
⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆
The software school I go to was troubleshooting a student's issue with virtualization saying it was disabled when trying to run Ubuntu on WSL and would fail saying virtualization was not enabled, even though it was showing in task manager and other ways we verified it was enabled. When we ran `bcdedit` we found that hypervisionlaunchtype was set to off by default. We solved the issue by setting it to auto and restarting. This is the answer!

**Comment by Nachtalb** [comment]
Someone from spacedesk over at the spacedesk forum posted a potential fix, I can't try it anymore as I reset have reset my PC in the meantime. I hope it can help someone else. 
> Currently it seems as if there could be a side effect with microsoft hyper-v virtualization infrastructure driver.
It seems as if the problem could be solved by manually re-installing this driver.
Please try the fix below and let us know if it solved the problem.
> 1. Go to device manager -> Action -> add legacy hardware.
> 2. Choose Option „Select hardware manual from list“ in the wizard
> 3. Choose „Systemdevices“ in the list
> 4. Choose „Microsoft“ in vendors
> 5. Choose „microsoft hyper-v virtualization infrastructure driver“ in models
> 6. Follow the wizard to finish the installation
> 7. Restart the computer
>
> ![image](https://user-images.githubusercontent.com/9467802/148312181-0126bbcc-112c-4a93-b417-18d742fb8ac8.png)

**Comment by abhishekadhikari23** [comment]
> Someone from spacedesk over at the spacedesk forum posted a potential fix, I can't try it anymore as I have reset my PC in the meantime. I hope it can help someone else.
> 
> > Currently it seems as if there could be a side effect with microsoft hyper-v virtualization infrastructure driver.
> > It seems as if the problem could be solved by manually re-installing this driver.
> > Please try the fix below and let us know if it solved the problem.
> > 
> > 1. Go to device manager -> Action -> add legacy hardware.
> > 2. Choose Option „Select hardware manual from list“ in the wizard
> > 3. Choose „Systemdevices“ in the list
> > 4. Choose „Microsoft“ in vendors
> > 5. Choose „microsoft hyper-v virtualization infrastructure driver“ in models
> > 6. Follow the wizard to finish the installation
> > 7. Restart the computer
> > 
> > ![image](https://user-images.githubusercontent.com/9467802/148312181-0126bbcc-112c-4a93-b417-18d742fb8ac8.png)

This surely works mates. Saved my life. Had this problem in past. Did a clean install of Windows several times but this problem used to come back. But this saved me.

**Comment by ElCosmoXD** [comment]
> Someone from spacedesk over at the spacedesk forum posted a potential fix, I can't try it anymore as I have reset my PC in the meantime. I hope it can help someone else.
> 
> > Currently it seems as if there could be a side effect with microsoft hyper-v virtualization infrastructure driver.
> > It seems as if the problem could be solved by manually re-installing this driver.
> > Please try the fix below and let us know if it solved the problem.
> > 
> > 1. Go to device manager -> Action -> add legacy hardware.
> > 2. Choose Option „Select hardware manual from list“ in the wizard
> > 3. Choose „Systemdevices“ in the list
> > 4. Choose „Microsoft“ in vendors
> > 5. Choose „microsoft hyper-v virtualization infrastructure driver“ in models
> > 6. Follow the wizard to finish the installation
> > 7. Restart the computer
> > 
> > ![image](https://user-images.githubusercontent.com/9467802/148312181-0126bbcc-112c-4a93-b417-18d742fb8ac8.png)

I tested this and it solved the problem, Thanks.

![image](https://user-images.githubusercontent.com/37759352/148698154-ca99d32c-2b88-4cb6-8001-891ce816a202.png)

**Comment by yasharya2901** [comment]
> Someone from spacedesk over at the spacedesk forum posted a potential fix, I can't try it anymore as I have reset my PC in the meantime. I hope it can help someone else.
> 
> > Currently it seems as if there could be a side effect with microsoft hyper-v virtualization infrastructure driver.
> > It seems as if the problem could be solved by manually re-installing this driver.
> > Please try the fix below and let us know if it solved the problem.
> > 
> > 1. Go to device manager -> Action -> add legacy hardware.
> > 2. Choose Option „Select hardware manual from list“ in the wizard
> > 3. Choose „Systemdevices“ in the list
> > 4. Choose „Microsoft“ in vendors
> > 5. Choose „microsoft hyper-v virtualization infrastructure driver“ in models
> > 6. Follow the wizard to finish the installation
> > 7. Restart the computer
> > 
> > ![image](https://user-images.githubusercontent.com/9467802/148312181-0126bbcc-112c-4a93-b417-18d742fb8ac8.png)

Thanks, this really worked for me. Now I can use WSL without any problems.

**Comment by Gauravkumar2701** [comment]
This bcdedit /set hypervisorlaunchtype auto is not working in windows 11

**Comment by bykevinyang** [comment]
> > Someone from spacedesk over at the spacedesk forum posted a potential fix, I can't try it anymore as I have reset my PC in the meantime. I hope it can help someone else.
> > > Currently it seems as if there could be a side effect with microsoft hyper-v virtualization infrastructure driver.
> > > It seems as if the problem could be solved by manually re-installing this driver.
> > > Please try the fix below and let us know if it solved the problem.
> > > 
> > > 1. Go to device manager -> Action -> add legacy hardware.
> > > 2. Choose Option „Select hardware manual from list“ in the wizard
> > > 3. Choose „Systemdevices“ in the list
> > > 4. Choose „Microsoft“ in vendors
> > > 5. Choose „microsoft hyper-v virtualization infrastructure driver“ in models
> > > 6. Follow the wizard to finish the installation
> > > 7. Restart the computer
> > > 
> > > ![image](https://user-images.githubusercontent.com/9467802/148312181-0126bbcc-112c-4a93-b417-18d742fb8ac8.png)
> 
> This surely works mates. Saved my life. Had this problem in past. Did a clean install of Windows several times but this problem used to come back. But this saved me.

BTW this solution references this [forum ](https://www.spacedesk.net/forums/topic/wsl2-not-starting-after-spacedesk-installation/page/2/). After installing "microsoft hyper-v virtualization infrastructure driver" from the legacy hardware wizard, it gave me an error that it the infrastrucutre driver could not start. 

To fix this, I followed what one of the forum users posted:

> In my situation,
> – After reinstall the microsoft hyper-v virtualization infrastructure driver,
> – In the device manager list, it still shows that there is a problem with this driver，
> – Then i uninstall it in the device manager, After automatic reinstallation

Credit to Spacedesk forum user Aiken

**Comment by ReactiveSlime** [comment]
Found on the spacedeak forums and worked for me

Currently it seems as if there could be a side effect with microsoft hyper-v virtualization infrastructure driver.
It seems as if the problem could be solved by manually re-installing this driver.
Please try the fix below and let us know if it solved the problem.
1. Go to device manager -> Action -> add legacy hardware.
2. Choose Option „Select hardware manual from list“ in the wizard
3. Choose „Systemdevices“ in the list
4. Choose „Microsoft“ in vendors
5. Choose „microsoft hyper-v virtualization infrastructure driver“ in models
6. Follow the wizard to finish the installation
7. Restart the computer

**Comment by tijko** [comment]
> I was having this same problem. I disabled AMD SVM, booted, then re-enabled AMD SVM and it worked again. Hope this helps

I don't feel great pointing to this but this is exactly what I did and it fixed the issue.

Between Microsoft, Docker, and the OEM the group most likely needs some collaborative debugging.

**Comment by tijko** [comment]
> > Someone from spacedesk over at the spacedesk forum posted a potential fix, I can't try it anymore as I have reset my PC in the meantime. I hope it can help someone else.
> 
> > 
> 
> > > Currently it seems as if there could be a side effect with microsoft hyper-v virtualization infrastructure driver.
> 
> > > It seems as if the problem could be solved by manually re-installing this driver.
> 
> > > Please try the fix below and let us know if it solved the problem.
> 
> > > 
> 
> > > 1. Go to device manager -> Action -> add legacy hardware.
> 
> > > 2. Choose Option „Select hardware manual from list“ in the wizard
> 
> > > 3. Choose „Systemdevices“ in the list
> 
> > > 4. Choose „Microsoft“ in vendors
> 
> > > 5. Choose „microsoft hyper-v virtualization infrastructure driver“ in models
> 
> > > 6. Follow the wizard to finish the installation
> 
> > > 7. Restart the computer
> 
> > > 
> 
> > > ![image](https://user-images.githubusercontent.com/9467802/148312181-0126bbcc-112c-4a93-b417-18d742fb8ac8.png)
> 
> 
> 
> Thanks, this really worked for me. Now I can use WSL without any problems.

I haven't had a chance to do a test with this fix but I can say that I have been able to switch it off (actually off) in the "virtualization" section in the UEFI for a Dell XPS I have.

While not ideal, this exact situation has occurred on two identical but separate laptops for my employer.  After turning off the WSL feature both laptops (same Windows 10 Enterprise versions) will then incorrectly report that the VMM instruction set are not enabled.  Going into UEFI, as I said, turning it off properly->reboot->shutdown->re-enable-> all is working as before.

**Comment by github-wow** [comment]
> WSL : bcdedit /set hypervisorlaunchtype auto + RESTART
> VirtualBox: bcdedit /set hypervisorlaunchtype off + RESTART

worked for me

**Comment by LynxTR** [comment]
After uninstalling spacedesk and rebooting, wsl seems like working again!

**Comment by LeonBraga** [comment]
> I was having this same problem. I disabled AMD SVM, booted, then re-enabled AMD SVM and it worked again. Hope this helps

I was with the same problem! Doing this fixed it for me, thank you very much @Gareth001

**Comment by ddimick** [comment]
> I was having this same problem. I disabled AMD SVM, booted, then re-enabled AMD SVM and it worked again. Hope this helps

I have been beating my head against this wall for a long time, and this did the trick.

**Comment by elevenTrue** [comment]
> Someone from spacedesk over at the spacedesk forum posted a potential fix, I can't try it anymore as I have reset my PC in the meantime. I hope it can help someone else.
> 
> > Currently it seems as if there could be a side effect with microsoft hyper-v virtualization infrastructure driver.
> > It seems as if the problem could be solved by manually re-installing this driver.
> > Please try the fix below and let us know if it solved the problem.
> > 
> > 1. Go to device manager -> Action -> add legacy hardware.
> > 2. Choose Option „Select hardware manual from list“ in the wizard
> > 3. Choose „Systemdevices“ in the list
> > 4. Choose „Microsoft“ in vendors
> > 5. Choose „microsoft hyper-v virtualization infrastructure driver“ in models
> > 6. Follow the wizard to finish the installation
> > 7. Restart the computer
> > 
> > ![image](https://user-images.githubusercontent.com/9467802/148312181-0126bbcc-112c-4a93-b417-18d742fb8ac8.png)


Thank you very much. I installed spacedesk to cause this problem. It seems that you can't install software randomly

**Comment by robertgro** [comment]
Still an issue people run into.

**Comment by benhichem** [comment]
> > Someone from spacedesk over at the spacedesk forum posted a potential fix, I can't try it anymore as I have reset my PC in the meantime. I hope it can help someone else.
> > > Currently it seems as if there could be a side effect with microsoft hyper-v virtualization infrastructure driver.
> > > It seems as if the problem could be solved by manually re-installing this driver.
> > > Please try the fix below and let us know if it solved the problem.
> > > 
> > > 1. Go to device manager -> Action -> add legacy hardware.
> > > 2. Choose Option „Select hardware manual from list“ in the wizard
> > > 3. Choose „Systemdevices“ in the list
> > > 4. Choose „Microsoft“ in vendors
> > > 5. Choose „microsoft hyper-v virtualization infrastructure driver“ in models
> > > 6. Follow the wizard to finish the installation
> > > 7. Restart the computer
> > > 
> > > ![image](https://user-images.githubusercontent.com/9467802/148312181-0126bbcc-112c-4a93-b417-18d742fb8ac8.png)
> 
> Thanks, this really worked for me. Now I can use WSL without any problems.



this worked for me amazing still don't know what that happen by yeah

**Comment by pbies** [comment]
For me, Windows 10, it works like that:

When there is "bcdedit /set hypervisorlaunchtype auto" WSL2 works ok, but VirtualBox guests are very slow.

When there is "bcdedit /set hypervisorlaunchtype off" WSL2 does not work (error message about missing Hyper-V), but VirtualBox guests are running full speed.

Both after host restarts.

Installing mentioned driver didn't help at all.
